gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -169,12 +196,44 @@ public void run() {
         throw new CopyNotSupportedException(schema + " is not copyable");
       }
 
+      if (isContinuousTask())
+      {
+        Extractor underlyingExtractor = this.taskContext.getExtractor();
+        if (!(underlyingExtractor instanceof StreamingExtractor)) {
+          LOG.error("Extractor {}  is not an instance of StreamingExtractor but the task is configured to run in continuous mode", underlyingExtractor.getClass().getName());
+          throw new TaskInstantiationException("Extraction " + underlyingExtractor.getClass().getName()
+              + " is not an instance of StreamingExtractor but the task is configured to run in continuous mode");
+        }
+        if (!(underlyingExtractor instanceof WatermarkStorage)) {
+          LOG.error("Extractor {}  is not an instance of WatermarkStorage but the task is configured to run in continuous mode", underlyingExtractor.getClass().getName());
+          throw new TaskInstantiationException("Extractor " + underlyingExtractor.getClass().getName()
+              + " is not an instance of WatermarkStorage but the task is configured to run in continuous mode");
+        }
+        long commitIntervalMillis = 1000; // TODO: Configure
+        this.watermarkManager = Optional.of(this.closer.register
```

Comment:
Commenting here since github isn't letting me comment on commit() method since you haven't touched it :)  : 
- We should tweak commit() to consolidate handle watermark update for last of writes 
- Revisit publishTaskData() being invoked from commit()